purchase_sale_stock_inter_company/models/stock_move_line.py

```diff
@@ -0,0 +1,95 @@
+from odoo import api, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_move_lines = super().create(vals_list)
+        for move_line in new_move_lines:
+            purchase_line_origin = (
+                move_line.move_id.sale_line_id.sudo().auto_purchase_line_id
+            )
+            if move_line.state == "done" and purchase_line_origin:
+                po_moves = purchase_line_origin.move_ids.filtered(
+                    lambda m: m.picking_id
+                    == move_line.move_id.picking_id.intercompany_picking_id
+                )
+                if not po_moves:
+                    continue
+                po_move_line_vals = po_moves._prepare_move_line_vals()
```

Comment:
Why not simply copying this sml values and changing the `lot`? Anyway, I think there's an argument for passing the quantity directly to the method.

Reply:
And on write, you may need also to create some sml. Why not unifying the code?

Reply:
The argument `quantity` is used to populate the `product_uom_qty` field. In this case, I need to populate the `qty_done` field.
[Reference](https://github.com/odoo/odoo/blob/4cc086d330b73514fbc64a7fcf22d8a7a9f1b691/addons/stock/models/stock_move.py#L1105).

In the `write` method, the corresponding `stock.move.line` in the other company is always updated. There's no need to create a new record. Note that this code applies only to moves in the done state.

Reply:
And shouldn't you populate also that field?

Reply:
Done